app/v2/services/RetrieveBalanceAndTransactionsService.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.services
+
+import api.controllers.EndpointLogContext
+import api.models.errors._
+import api.models.outcomes.ResponseWrapper
+import cats.data.EitherT
+import uk.gov.hmrc.http.HeaderCarrier
+import utils.Logging
+import v2.connectors.RetrieveBalanceAndTransactionsConnector
+import v2.models.request.retrieveBalanceAndTransactions.RetrieveBalanceAndTransactionsRequest
+import v2.models.response.retrieveBalanceAndTransactions.RetrieveBalanceAndTransactionsResponse
+import v2.services.RetrieveBalanceAndTransactionsService.downstreamErrorMap
+import v2.support.DownstreamResponseMappingSupport
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class RetrieveBalanceAndTransactionsService @Inject()(connector: RetrieveBalanceAndTransactionsConnector) extends
+  DownstreamResponseMappingSupport with Logging {
+
+  def retrieveBalanceAndTransactions(request: RetrieveBalanceAndTransactionsRequest)(implicit
+                                                                                     hc: HeaderCarrier,
+                                                                                     ec: ExecutionContext,
+                                                                                     logContext: EndpointLogContext,
+                                                                                     correlationId: String): Future[Either[ErrorWrapper, ResponseWrapper[RetrieveBalanceAndTransactionsResponse]]] = {
+
+    val result = for {
+      desResponseWrapper <- EitherT(connector.retrieveBalanceAndTransactions(request)).leftMap(mapDownstreamErrors(downstreamErrorMap))
+    } yield desResponseWrapper.map(des => des)
```

Comment:
The for comprehension and the `map(des => des)` don't modify the data, so can be replace with

```
val result = EitherT(connector.retrieveBalanceAndTransactions(request)).leftMap(mapDownstreamErrors(downstreamErrorMap))
```